conda/core/python_dist.py

```diff
@@ -0,0 +1,1026 @@
+    information which is assumed to be *correct*. As such no errors are raised.
+
+    Example
+    -------
+    PySpec(name='requests', extras=['security'], constraints='>=3.3.0',
+           marker='foo >= 2.7 or bar == 1', url=''])
+    """
+    name, extras, const = spec, [], ''
+
+    # Remove excess whitespace
+    spec = ' '.join(p for p in spec.split(' ') if p).strip()
+
+    # Extract marker (Assumes that there can only be one ';' inside the spec)
+    spec, marker = split_spec(spec, ';')
+
+    # Extract url (Assumes that there can only be one '@' inside the spec)
+    spec, url = split_spec(spec, '@')
+
+    # Find name, extras and constraints
+    r = PARTIAL_PYPI_SPEC_PATTERN.match(spec)
+    if r:
+        # Normalize name
+        name = r.group('name')
+        name = norm_package_name(name)  # TODO: Do we want this or not?
+
+        # Clean extras
+        extras = r.group('extras')
+        extras = [e.strip() for e in extras.split(',') if e] if extras else []
+
+        # Clean constraints
+        const = r.group('constraints')
+        const = ''.join(c for c in const.split(' ') if c).strip()
+        if const.startswith('(') and const.endswith(')'):
+            # Remove parens
+            const = const[1:-1]
+
+    return PySpec(name=name, extras=extras, constraints=const, marker=marker, url=url)
+
+
+def get_conda_anchor_files_and_records(python_records):
+    """Return the anchor files for the conda records of python packages."""
+    anchor_file_endings = ('.egg-info/PKG-INFO', '.dist-info/RECORD', '.egg-info')
+    conda_python_packages = odict()
+
+    for prefix_record in python_records:
+        for fpath in prefix_record.files:
+            if fpath.endswith(anchor_file_endings) and 'site-packages' in fpath:
+                # Then 'fpath' is an anchor file
+                conda_python_packages[fpath] = prefix_record
+
+    return conda_python_packages
+
+
+def get_site_packages_anchor_files(site_packages_path, site_packages_dir):
+    """Get all the anchor files for the site packages directory."""
+    site_packages_anchor_files = set()
+    for fname in listdir(site_packages_path):
+        if fname.endswith('.dist-info'):
+            anchor_file = "%s/%s/%s" % (site_packages_dir, fname, 'RECORD')
+        elif fname.endswith(".egg-info"):
+            if isfile(join(site_packages_path, fname)):
+                anchor_file = "%s/%s" % (site_packages_dir, fname)
+            else:
+                anchor_file = "%s/%s/%s" % (site_packages_dir, fname, "PKG-INFO")
+        elif fname.endswith(".egg"):
+            if isdir(join(site_packages_path, fname)):
+                anchor_file = "%s/%s/%s/%s" % (site_packages_dir, fname, "EGG-INFO", "PKG-INFO")
+            # FIXME: If it is a .egg file, we need to unzip the content to be
+            # able. Do this once and leave the directory, and remove the egg
+            # (which is a zip file in disguise?)
+        elif fname.endswith('.egg-link'):
```

Comment:
When installing using

`python setup.py install`, a new type of folder is created (<>.egg) and <>.egg files, which are zipped. How should we handle these?

I would like to unzip the file and leave it unzipped, and remove/rename? the original .egg file. 

Thoughts?